regulations/static/components/tooltips/CopyBtn.vue

```diff
@@ -0,0 +1,259 @@
+            <p class="citation-title">{{ this.formatted_citation }}</p>
+            <div class="action-btns">
+                <ActionBtn
+                    @action-btn-click="handleActionClick"
+                    :selectedAction="selectedAction"
+                    :status="copyStatus"
+                    actionType="link"
+                ></ActionBtn>
+                <ActionBtn
+                    @action-btn-click="handleActionClick"
+                    :selectedAction="selectedAction"
+                    :status="copyStatus"
+                    actionType="citation"
+                ></ActionBtn>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+import ActionBtn from "./ActionBtn.vue";
+
+const getAnchorY = (el, elType) => {
+    if (!el) return 0;
+
+    return elType === "labeled-icon"
+        ? el.offsetWidth / 2
+        : el.offsetWidth * 0.7;
+};
+
+const getAnchorX = (el, elType) => {
+    if (!el) return 0;
+
+    return parseInt(window.getComputedStyle(el).fontSize) + 20;
+};
+
+const appendPxSuffix = (int) => `${int}px`;
+
+const leftWarning = (el) => el.getBoundingClientRect().left < 130;
+
+export default {
+    name: "copy-btn",
+
+    components: {
+        ActionBtn,
+    },
+
+    props: {
+        btn_type: {
+            type: String,
+            required: true,
+        },
+        title: {
+            type: String,
+            required: true,
+        },
+        hash: {
+            type: String,
+            required: true,
+        },
+        formatted_citation: {
+            type: String,
+            required: true,
+        },
+    },
+
+    data() {
+        return {
+            entered: false,
+            clicked: false,
+            leftSafe: true,
+            anchorY: 0,
+            anchorX: 0,
+            label: "Copy Link or Citation",
+            selectedAction: null,
+            copyStatus: "idle",
+        };
+    },
+
+    computed: {
+        ariaLabel() {
+            return `${this.label} for ${this.title}`;
+        },
+        buttonClasses() {
+            return {
+                "copy-btn-labeled": this.btn_type === "labeled-icon",
+            };
+        },
+        tooltipClasses() {
+            return {
+                "tooltip-caret": this.leftSafe,
+                "tooltip-caret-left": !this.leftSafe,
+            };
+        },
+        tooltipStyles() {
+            return {
+                left: this.anchorY,
+                bottom: this.anchorX,
+                transform: `translate(-${this.leftSafe ? 50 : 20}%, 0)`,
+            };
+        },
+    },
+
+    watch: {
+        copyStatus: async function (newStatus, oldStatus) {
+            if (
+                newStatus === "pending" &&
+                (oldStatus === "idle" || oldStatus === "success")
+            ) {
+                try {
+                    // async write to clipboard
+                    await navigator.clipboard.writeText(this.getCopyText());
```

Comment:
Do we need to support IE?  It does not support this api  https://developer.mozilla.org/en-US/docs/Web/API/Navigator/clipboard